### DISCUSSION TITLE
How to create iOS Notification Service Extension?

### DISCUSSION THREAD

**Discussion by Lyfei** [apache/cordova#373] [Q&A] [answered]
I had a  ionic project and used cordova created  iOS native project，but how to create Notification Service Extension together? The relevant information is not found in the document。The purpose is to broadcast the push message by voice。

**Comment by breautek** [comment, marked as answer]
I can't really provide any advice on creating a notification service specifically but anything that requires extending the native platform is going to require a Cordova plugin. A cordova plugin allows you to build on top of the native code. Creating an iOS plugin would be a good place to start, and the [docs](https://cordova.apache.org/docs/en/11.x/guide/platforms/ios/plugin.html) shows you how to build a simple plugin that you can extend from.
